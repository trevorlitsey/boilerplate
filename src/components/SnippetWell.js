import React from 'react';
import styled from 'styled-components';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faPlusSquare from '@fortawesome/fontawesome-free-solid/faPlusSquare';
import { Tooltip, OverlayTrigger } from 'react-bootstrap';

const Container = styled.div`
	display: flex;
	flex-wrap: wrap;
	background: var(--light);
	padding: 10px;
	margin-bottom: 20px;

	& > .card {
		display: grid;
		grid-template-columns: 1fr 3fr;
		align-items: center;
		grid-gap: 6px;
		padding: 8px;
		margin: 2px;

		
	}

	.icon:hover {
		cursor: pointer;
	}
`

const H4 = styled.h3`
	text-decoration: underline;
	margin-bottom: .75rem;
`

const tooltip = (
	<div className="tooltip show bs-tooltip-top" role="tooltip">
		<div className="arrow"></div>
		<div className="tooltip-inner">
			Some tooltip text!
  	</div>
	</div>
)

const Card = ({ title, tags }) => (
	<div className="card">
		<FontAwesomeIcon icon={faPlusSquare} className="icon" />
		<OverlayTrigger placement="top" overlay={tooltip}>
			<p className="card-text">{title}</p>
		</OverlayTrigger>
	</div>
)

class SnippetWell extends React.PureComponent {

	render() {

		const { snippets } = this.props;

		return (
			<div>
				<H4>Snippets</H4>
				<Container className="rounded">
					{snippets.map(snippet => <Card key={snippet.id} {...snippet} />)}
				</Container>
			</div>
		)
	}
}

export default SnippetWell;