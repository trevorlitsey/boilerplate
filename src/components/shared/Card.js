import React from 'react';
import { string, object, bool } from 'prop-types';
import { Link } from 'react-router-dom';

const grid = 8;

const getStyles = (overrideStyles, isActive) => ({
	userSelect: 'none',
	padding: grid * 2,
	lineHeight: 1,
	margin: `0 0 ${grid}px ${grid}px`,
	background: isActive ? 'var(--primary)' : '',
	color: isActive ? 'var(--white)' : '',

	...overrideStyles,
})

const Tag = ({ value }) => (
	<span className="badge badge-light">{value}</span>
)

const Card = ({ title, text, tags, id, overrideStyles, isActive, showModal }) => (
	<div className="card" style={getStyles(overrideStyles, isActive)}>
		<h5 className="card-title">{title}</h5>
		<p className="card-text">{text.length > 80 ? text.substring(0, 80) + ' ...' : text}</p>
		<div style={{ marginTop: 'auto', display: 'flex', justifyContent: 'space-between' }}>
			<Link to="#" style={{ marginRight: 12, marginTop: 'auto' }} onClick={() => showModal(id)}>edit</Link>
			<div>
				{tags && tags.map(tag => <Tag key={tag.id} value={tag.value} />)}
			</div>
		</div>
	</div>
)

Card.propTypes = {
	title: string.isRequired,
	text: string.isRequired,
	overrideStyles: object,
	isActive: bool,
}

export default Card;