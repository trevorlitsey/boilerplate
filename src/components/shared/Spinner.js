import React from 'react';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faSpinner from '@fortawesome/fontawesome-free-solid/faSpinner';
import styled from 'styled-components';

const Wrapper = styled.div`
	.icon {
		display: flex;
		min-width: 50px;
		margin: auto;
		font-size: 2rem;
		color: var(--blue);
		opacity: .6;
	}
`

const Spinner = () => (
	<Wrapper>
		<FontAwesomeIcon className="icon" icon={faSpinner} spin />
	</Wrapper>
)

// no props

export default Spinner;