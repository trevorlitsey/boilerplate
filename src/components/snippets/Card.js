import React from 'react';
import { string, object, bool, array, func } from 'prop-types';
import { Link } from 'react-router-dom';

const grid = 8;

const styles = {
	userSelect: 'none',
	padding: grid * 2,
	lineHeight: 1,
}

export const Tag = ({ value }) => (
	<span className="badge badge-light">{value}</span>
)

const Card = ({ title, text, tags, id, showModal }) => (
	<div className="card" style={styles}>
		<h5 className="card-title">{title}</h5>
		<p className="card-text">{text.length > 80 ? text.substring(0, 80) + ' ...' : text}</p>
		<div style={{ marginTop: 'auto', display: 'flex', justifyContent: 'space-between' }}>
			<a href="#" style={{ marginRight: 12, marginTop: 'auto' }} onClick={() => showModal(null, id)}>edit</a>
			<div>
				{tags && tags.map(tag => <Tag key={tag.id} value={tag.value} />)}
			</div>
		</div>
	</div>
)

Card.propTypes = {
	title: string.isRequired,
	text: string.isRequired,
	tags: array,
	id: string.isRequired,
	showModal: func.isRequired,
}

export default Card;