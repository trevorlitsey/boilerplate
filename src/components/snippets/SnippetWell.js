import React from 'react';
import styled from 'styled-components';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faArrowDown from '@fortawesome/fontawesome-free-solid/faArrowDown';

const Container = styled.div`
	width: 100%;
	min-height: 200px;
	background: var(--light);
	margin-bottom: 10px;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	color: var(--secondary);
	box-shadow: 0px 0px 2px inset;
`

const SnippetWell = ({ snippets }) => {

	if (!snippets) {
		return (
			<Container className="rounded">
				<h3>No Snippets Yet!</h3>
				<p>click the button below to get started</p>
				<FontAwesomeIcon icon={faArrowDown} />
			</Container>
		)
	}

	return (
		<div className="container-fluid">

		</div>
	)
}

export default SnippetWell;