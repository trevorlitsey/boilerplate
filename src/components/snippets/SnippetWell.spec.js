import React from 'react';
import { shallow } from 'enzyme';

import Spinner from '../shared/Spinner';
import Card from './Card';
import SnippetWell from './SnippetWell';

describe('SnippetWell', () => {

	it('should render without crashing', () => {
		const wrapper = renderSnippetWell();
		expect(wrapper).toMatchSnapshot();
	})

	it('should render spinner when loading', () => {
		const wrapper = renderSnippetWell({ loading: true });
		expect(wrapper.find(Spinner).length).toBe(1);
	})

	it('should let user know there are no snippets if none provided', () => {
		const wrapper = renderSnippetWell({ snippets: {} });
		expect(wrapper.find('h3').text()).toEqual('No Snippets Yet!');
	})

	it('should render correct number of snippets provided', () => {
		const snippets = getSnippets();
		const wrapper = renderSnippetWell({ snippets });
		expect(wrapper.find(Card).length).toEqual(Object.keys(snippets).length)
	})

	it('should filter snippets according to filterQuery', () => {
		const filterQuery = 'section two';
		const wrapper = renderSnippetWell({ filterQuery });
		expect(wrapper.find(Card).length).toEqual(1)
	})

})

// -------
function renderSnippetWell(props = {}) {
	const propsToUser = {
		snippets: getSnippets(),
		showModal: () => { },
		filterQuery: '',
		loading: false,
		...props,
	}

	return shallow(<SnippetWell {...propsToUser} />)
}

function getSnippets() {
	const snippets = {
		jghgygc9: {
			title: 'Section Two',
			text: 'Praesent aliquet placerat augue in varius. Sed lorem metus, gravida at lectus nec, egestas convallis ex. Suspendisse accumsan quam nec laoreet fermentum. Donec scelerisque in lorem non aliquet. Nunc vestibulum eleifend felis eget dignissim. Integer ullamcorper ipsum ac imperdiet varius. Maecenas cursus egestas neque, quis placerat enim dignissim eu. Integer ultrices aliquam metus eget faucibus. Vivamus lobortis turpis odio, sed sodales odio elementum nec. Mauris facilisis ipsum et ante porta accumsan. Quisque molestie maximus leo, id cursus odio porttitor mattis. Quisque eu porttitor justo. Ut facilisis pretium arcu, eget mollis sem mattis ut.',
			tags: [
				{ id: 'jgtv2988', label: 'lots', value: 'lots' },
				{ id: 'jgtv29wa', label: 'of', value: 'of' },
				{ id: 'jgtv1e21', label: 'tags', value: 'tags' },
			]
		},
		jgux6edn: {
			title: 'Section One',
			text: 'this is the first section',
			tags: [],
		},
		jguxf292: {
			title: 'Section Three',
			text: 'this is the third section',
			tags: [],
		},
		jgxvm7zr: {
			tags: [],
			text: 'this is the fourth section',
			title: 'Section Four!',
		}
	}
	return snippets;
}