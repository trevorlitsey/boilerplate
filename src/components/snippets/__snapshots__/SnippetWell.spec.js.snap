// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnippetWell should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SnippetWell
    filterQuery=""
    loading={false}
    showModal={[Function]}
    snippets={
      Object {
        "jghgygc9": Object {
          "tags": Array [
            Object {
              "id": "jgtv2988",
              "label": "lots",
              "value": "lots",
            },
            Object {
              "id": "jgtv29wa",
              "label": "of",
              "value": "of",
            },
            Object {
              "id": "jgtv1e21",
              "label": "tags",
              "value": "tags",
            },
          ],
          "text": "Praesent aliquet placerat augue in varius. Sed lorem metus, gravida at lectus nec, egestas convallis ex. Suspendisse accumsan quam nec laoreet fermentum. Donec scelerisque in lorem non aliquet. Nunc vestibulum eleifend felis eget dignissim. Integer ullamcorper ipsum ac imperdiet varius. Maecenas cursus egestas neque, quis placerat enim dignissim eu. Integer ultrices aliquam metus eget faucibus. Vivamus lobortis turpis odio, sed sodales odio elementum nec. Mauris facilisis ipsum et ante porta accumsan. Quisque molestie maximus leo, id cursus odio porttitor mattis. Quisque eu porttitor justo. Ut facilisis pretium arcu, eget mollis sem mattis ut.",
          "title": "Section Two",
        },
        "jgux6edn": Object {
          "tags": Array [],
          "text": "this is the first section",
          "title": "Section One",
        },
        "jguxf292": Object {
          "tags": Array [],
          "text": "this is the third section",
          "title": "Section Three",
        },
        "jgxvm7zr": Object {
          "tags": Array [],
          "text": "this is the fourth section",
          "title": "Section Four!",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Card
          id="jghgygc9"
          showModal={[Function]}
          tags={
            Array [
              Object {
                "id": "jgtv2988",
                "label": "lots",
                "value": "lots",
              },
              Object {
                "id": "jgtv29wa",
                "label": "of",
                "value": "of",
              },
              Object {
                "id": "jgtv1e21",
                "label": "tags",
                "value": "tags",
              },
            ]
          }
          text="Praesent aliquet placerat augue in varius. Sed lorem metus, gravida at lectus nec, egestas convallis ex. Suspendisse accumsan quam nec laoreet fermentum. Donec scelerisque in lorem non aliquet. Nunc vestibulum eleifend felis eget dignissim. Integer ullamcorper ipsum ac imperdiet varius. Maecenas cursus egestas neque, quis placerat enim dignissim eu. Integer ultrices aliquam metus eget faucibus. Vivamus lobortis turpis odio, sed sodales odio elementum nec. Mauris facilisis ipsum et ante porta accumsan. Quisque molestie maximus leo, id cursus odio porttitor mattis. Quisque eu porttitor justo. Ut facilisis pretium arcu, eget mollis sem mattis ut."
          title="Section Two"
        />,
        <Card
          id="jgux6edn"
          showModal={[Function]}
          tags={Array []}
          text="this is the first section"
          title="Section One"
        />,
        <Card
          id="jguxf292"
          showModal={[Function]}
          tags={Array []}
          text="this is the third section"
          title="Section Three"
        />,
        <Card
          id="jgxvm7zr"
          showModal={[Function]}
          tags={Array []}
          text="this is the fourth section"
          title="Section Four!"
        />,
      ],
      "className": "rounded",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "jghgygc9",
        "nodeType": "function",
        "props": Object {
          "id": "jghgygc9",
          "showModal": [Function],
          "tags": Array [
            Object {
              "id": "jgtv2988",
              "label": "lots",
              "value": "lots",
            },
            Object {
              "id": "jgtv29wa",
              "label": "of",
              "value": "of",
            },
            Object {
              "id": "jgtv1e21",
              "label": "tags",
              "value": "tags",
            },
          ],
          "text": "Praesent aliquet placerat augue in varius. Sed lorem metus, gravida at lectus nec, egestas convallis ex. Suspendisse accumsan quam nec laoreet fermentum. Donec scelerisque in lorem non aliquet. Nunc vestibulum eleifend felis eget dignissim. Integer ullamcorper ipsum ac imperdiet varius. Maecenas cursus egestas neque, quis placerat enim dignissim eu. Integer ultrices aliquam metus eget faucibus. Vivamus lobortis turpis odio, sed sodales odio elementum nec. Mauris facilisis ipsum et ante porta accumsan. Quisque molestie maximus leo, id cursus odio porttitor mattis. Quisque eu porttitor justo. Ut facilisis pretium arcu, eget mollis sem mattis ut.",
          "title": "Section Two",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "jgux6edn",
        "nodeType": "function",
        "props": Object {
          "id": "jgux6edn",
          "showModal": [Function],
          "tags": Array [],
          "text": "this is the first section",
          "title": "Section One",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "jguxf292",
        "nodeType": "function",
        "props": Object {
          "id": "jguxf292",
          "showModal": [Function],
          "tags": Array [],
          "text": "this is the third section",
          "title": "Section Three",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "jgxvm7zr",
        "nodeType": "function",
        "props": Object {
          "id": "jgxvm7zr",
          "showModal": [Function],
          "tags": Array [],
          "text": "this is the fourth section",
          "title": "Section Four!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Card
            id="jghgygc9"
            showModal={[Function]}
            tags={
              Array [
                Object {
                  "id": "jgtv2988",
                  "label": "lots",
                  "value": "lots",
                },
                Object {
                  "id": "jgtv29wa",
                  "label": "of",
                  "value": "of",
                },
                Object {
                  "id": "jgtv1e21",
                  "label": "tags",
                  "value": "tags",
                },
              ]
            }
            text="Praesent aliquet placerat augue in varius. Sed lorem metus, gravida at lectus nec, egestas convallis ex. Suspendisse accumsan quam nec laoreet fermentum. Donec scelerisque in lorem non aliquet. Nunc vestibulum eleifend felis eget dignissim. Integer ullamcorper ipsum ac imperdiet varius. Maecenas cursus egestas neque, quis placerat enim dignissim eu. Integer ultrices aliquam metus eget faucibus. Vivamus lobortis turpis odio, sed sodales odio elementum nec. Mauris facilisis ipsum et ante porta accumsan. Quisque molestie maximus leo, id cursus odio porttitor mattis. Quisque eu porttitor justo. Ut facilisis pretium arcu, eget mollis sem mattis ut."
            title="Section Two"
          />,
          <Card
            id="jgux6edn"
            showModal={[Function]}
            tags={Array []}
            text="this is the first section"
            title="Section One"
          />,
          <Card
            id="jguxf292"
            showModal={[Function]}
            tags={Array []}
            text="this is the third section"
            title="Section Three"
          />,
          <Card
            id="jgxvm7zr"
            showModal={[Function]}
            tags={Array []}
            text="this is the fourth section"
            title="Section Four!"
          />,
        ],
        "className": "rounded",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "jghgygc9",
          "nodeType": "function",
          "props": Object {
            "id": "jghgygc9",
            "showModal": [Function],
            "tags": Array [
              Object {
                "id": "jgtv2988",
                "label": "lots",
                "value": "lots",
              },
              Object {
                "id": "jgtv29wa",
                "label": "of",
                "value": "of",
              },
              Object {
                "id": "jgtv1e21",
                "label": "tags",
                "value": "tags",
              },
            ],
            "text": "Praesent aliquet placerat augue in varius. Sed lorem metus, gravida at lectus nec, egestas convallis ex. Suspendisse accumsan quam nec laoreet fermentum. Donec scelerisque in lorem non aliquet. Nunc vestibulum eleifend felis eget dignissim. Integer ullamcorper ipsum ac imperdiet varius. Maecenas cursus egestas neque, quis placerat enim dignissim eu. Integer ultrices aliquam metus eget faucibus. Vivamus lobortis turpis odio, sed sodales odio elementum nec. Mauris facilisis ipsum et ante porta accumsan. Quisque molestie maximus leo, id cursus odio porttitor mattis. Quisque eu porttitor justo. Ut facilisis pretium arcu, eget mollis sem mattis ut.",
            "title": "Section Two",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "jgux6edn",
          "nodeType": "function",
          "props": Object {
            "id": "jgux6edn",
            "showModal": [Function],
            "tags": Array [],
            "text": "this is the first section",
            "title": "Section One",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "jguxf292",
          "nodeType": "function",
          "props": Object {
            "id": "jguxf292",
            "showModal": [Function],
            "tags": Array [],
            "text": "this is the third section",
            "title": "Section Three",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "jgxvm7zr",
          "nodeType": "function",
          "props": Object {
            "id": "jgxvm7zr",
            "showModal": [Function],
            "tags": Array [],
            "text": "this is the fourth section",
            "title": "Section Four!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
